pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java

```diff
@@ -454,25 +456,62 @@ private User upgradeTemporaryUser(Response response, UserDao userDao, User tempU
         return userDao.findUserByGuid(tempUserGuid).orElseThrow(() -> new DDPException("Could not find user with guid " + tempUserGuid));
     }
 
-    private void initProfilePreferredLanguage(Handle handle, User user, String auth0ClientCountryCode) {
-        Long languageId = handle.attach(JdbiCountry.class).getPrimaryLanguageIdByCountryCode(auth0ClientCountryCode);
+    private void initializeProfile(Handle handle, User user, UserRegistrationPayload payload) {
+        Long languageId = null;
+        Set<LanguageDto> supportedLanguages = handle.attach(StudyDao.class)
+                .findSupportedLanguagesByGuid(payload.getStudyGuid());
+        for (var language : supportedLanguages) {
+            if (language.getIsoCode().equalsIgnoreCase(payload.getLanguageCode())) {
+                languageId = language.getId();
+                break;
+            }
+        }
+
+        // If no language provided or language is not one of study's supported ones,
+        // then use study's default language. If all else fails, fallback to English.
         if (languageId == null) {
-            languageId = handle.attach(JdbiLanguageCode.class).getLanguageCodeId(EN_LANGUAGE_CODE);
+            languageId = supportedLanguages.stream()
+                    // .filter(lang -> lang.isDefault())     TODO: use the filter when ready
+                    .map(LanguageDto::getId)
+                    .findFirst()
+                    .orElseGet(() -> handle.attach(JdbiLanguageCode.class).getLanguageCodeId(EN_LANGUAGE_CODE));
```

Comment:
Consider refactoring the code in this method up to here to another called something like `determineLanguageId()`.